examples/35_gemm_softmax/softmax_epilogue.hpp

```diff
@@ -0,0 +1,299 @@
+
+    // Slice to get the tile this CTA is responsible for
+    auto [m_coord, n_coord, k_coord, l_coord] = blk_coord_mnkl;
+    Tensor gC = gC_mnl(_,_,m_coord,n_coord,l_coord);                                                 // (BLK_M,BLK_N)
+    Tensor gD = gD_mnl(_,_,m_coord,n_coord,l_coord);                                                 // (BLK_M,BLK_N)
+    Tensor gMax = gMax_mnl(_,m_coord,n_coord,l_coord);
+    Tensor gSum = gSum_mnl(_,m_coord,n_coord,l_coord);
+
+    //Represent the shared tensor
+    Tensor sC = make_tensor(make_smem_ptr(reinterpret_cast<ElementAccumulator*>(smem_buf)), make_layout(make_shape(M_tile, N_tile)));
+
+    // Partition the tiles to match the accumulator partitioning
+    auto thr_mma = tiled_mma.get_thread_slice(thread_idx);
+    Tensor tCgD = thr_mma.partition_C(gD);                                       // (VEC,THR_M,THR_N)
+    Tensor tCgC = thr_mma.partition_C(gC);                                       // (VEC,THR_M,THR_N)
+    Tensor tCsC = thr_mma.partition_C(sC);                                       // (VEC,THR_M,THR_N)
+
+    static_assert(is_static<FrgLayout>::value, "Accumulator layout must be static");
+    CUTE_STATIC_ASSERT_V(size(tCgC) == size(tCgD),
+        "Source and destination must have the same number of elements.");
+    CUTE_STATIC_ASSERT_V(size(tCgD) == size(accumulators),
+        "Accumulator count must have the same destination element count.");
+
+    // Make an identity coordinate tensor for predicating our output MN tile
+    auto cD = make_identity_tensor(make_shape(unwrap(shape<0>(gD)), unwrap(shape<1>(gD))));
+    Tensor tCcD = thr_mma.partition_C(cD);
+
+    if(is_source_needed()){
+      CUTLASS_PRAGMA_UNROLL
+      for (int i = 0; i < size<0>(accumulators); ++i) {
+        CUTLASS_PRAGMA_UNROLL
+        for (int j = 0; j < size<1>(accumulators); ++j) {
+          CUTLASS_PRAGMA_UNROLL
+          for (int k = 0; k < size<2>(accumulators); ++k) {
+            if (elem_less(tCcD(i,j,k), make_coord(get<0>(residue_mnk), get<1>(residue_mnk)))) {
+              accumulators(i,j,k) = epilogue_op(accumulators(i,j,k), tCgC(i,j,k));
+              tCgD(i,j,k) = accumulators(i,j,k);
+              tCsC(i,j,k) = accumulators(i,j,k);
+            }
+          }
+        }
+      }
+    } else{
+      CUTLASS_PRAGMA_UNROLL
+      for (int i = 0; i < size<0>(accumulators); ++i) {
+        CUTLASS_PRAGMA_UNROLL
+        for (int j = 0; j < size<1>(accumulators); ++j) {
+          CUTLASS_PRAGMA_UNROLL
+          for (int k = 0; k < size<2>(accumulators); ++k) {
+            if (elem_less(tCcD(i,j,k), make_coord(get<0>(residue_mnk), get<1>(residue_mnk)))) {
+              accumulators(i,j,k) = epilogue_op(accumulators(i,j,k));
+              tCgD(i,j,k) = accumulators(i,j,k);
+              tCsC(i,j,k) = accumulators(i,j,k);
+            } 
+          }
+        }
+      }
+    }
+
+    syncthreads();
+
+    // assumption: size<0>(sC) == wg size
```

Comment:
I don't see a nice way of doing it. I only get BlockShape in this call, so I can not do that outside. wg size on the other hand is a runtime value, so I can not use static_assert. And I would like to avoid (non-static) assert in the kernel for performance reasons.

Reply:
I would expect that runtime `assert`s would be stripped out of a Release build anyway, so shouldn't affect perf? Certainly there are a fair few `assert`s in the codebase.

`BlockShapeMNK` is static `(128, 128, 32)`. I can't quite work out what your wg size is, but I think its `(MaxThreadsPerBlock,1,1)` i.e. 1024? If so, I guess 7/8 of your threads aren't doing anything here? Maybe that's fine, but I don't think the assumption in the comment necessarily holds.

Also in the loop, you seem to be iterating up to `size<0>(sC)`  - shouldn't that be `size<1>(sC)`?

Reply:
`MaxThreadsPerBlock = CUTE_STATIC_V(cute::size(TiledMma{}));`. Somehow that gets us 128.

Added an assert and fixed iteration to `size<1>(sC)`.